internal/service/searchindex/resource_search_index.go

```diff
@@ -187,79 +187,78 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	projectID := ids["project_id"]
 	clusterName := ids["cluster_name"]
 	indexID := ids["index_id"]
+	indexName := d.Get("name").(string)
 
-	searchIndex, _, err := connV2.AtlasSearchApi.GetAtlasSearchIndex(ctx, projectID, clusterName, indexID).Execute()
+	if d.HasChange("name") || d.HasChange("type") || d.HasChange("database") || d.HasChange("collection_name") {
+		return diag.Errorf("error updating search index (%s): attributes name, type, database and collection_name can't be updated", indexName)
+	}
```

Comment:
👍 though in this case giving the user the option to recreate the index with RequiresReplace could help.

Reply:
as we a have a pending task to evaluate what to recreate or not, I preferred to err on the safe side here. I assume this will also be similar to the UI